goblin/provider.py

```diff
@@ -0,0 +1,11 @@
+class Provider:
+    """Superclass for provider plugins"""
+    DEFAULT_OP_ARGS = {}
+
+    @classmethod
+    def get_default_op_args(cls, processor):
+        return cls.DEFAULT_OP_ARGS.get(processor, dict())
+
+
+class TinkerGraph(Provider):  # TODO
+    """Default provider"""
```

Comment:
Syntactically, it doesn't have to be there because the doc string is an expression. Is the 'pass' required by PEP8?